### PR TITLE
Solve minor delete key bug Fix: 1232

### PR DESCRIPTION
**Pull request by harshitbansal05**
This pull request aims to solve #1232. It solves the minor error while deleting values from the `keywords` list.

### CODE REVIEW

**Comment by henrykironde** [comment]
@harshitbansal05 which python version are you using?

**Comment by henrykironde** [comment]
I think this was from the bug that Andrew is currently fixing because both methods do slightly the same thing.
pop calls `del`, but it first gets the value to be removed as the return deleted value
`removed = obj[key]; del obj[key]; return removed`
let me know what you think. We could alternatively wait till Andrew finishes up with fixing the bug and then retest this.

**Comment by harshitbansal05** [comment]
Hi, @henrykironde I was using python 3, but I think this problem is independent of the python version. IMO, the error is that the variable `del_key` is not defined in that scope. I think someone has accidentally written it in the place of `del_val`. Also, `pop` takes the index of the item to be deleted from the list, while `remove` takes the value as the argument, and deletes the first occurrence of that value. Reference: https://stackoverflow.com/questions/11520492/difference-between-del-remove-and-pop-on-lists/11520540

File: retriever/lib/datapackage.py [hunk 1]
```diff
@@ -356,7 +356,8 @@ def edit_dict(obj, tabwidth=0):
                         if del_val not in obj[key]:
                             print("Invalid value: Not found.")
                             continue
-                        print("Removed " + str(obj[key].pop(del_key)))
+                        obj[key].remove(del_val)
```
Comment (henrykironde) [inline retriever/lib/datapackage.py]: Can you check the class of obj using print(class(obj));
This seems like a dictionary not a list.

Comment (harshitbansal05) [reply retriever/lib/datapackage.py]: Yes, @henrykironde `obj` is a dictionary.

Comment (henrykironde) [reply retriever/lib/datapackage.py]: I do not think remove will work since it is a dictionary

Comment (harshitbansal05) [reply retriever/lib/datapackage.py]: @henrykironde but, `obj[key]` is a list, not a dictionary.

Comment (harshitbansal05) [reply retriever/lib/datapackage.py]: @henrykironde have you tried removing an item from the `keywords` list of a dataset? e.g, the value `plants` from the database iris.

Comment (henrykironde) [reply retriever/lib/datapackage.py]: @harshitbansal05, I have not tried running that part of code or tested it, I just read through the code. And it seems you are deleting something from a dictionary. I will have to run it and see what we are doing but it is clear that the initial value `del_key` was out of scope / not declared anywhere.

Comment (zhangcandrew) [reply retriever/lib/datapackage.py]: I think @harshitbansal05 has it right in this correction. obj is a dictionary of key->list mappings, so obj[key] will return a list. the pop function is used only for indices, so we should use remove to delete the value.

**Comment by zhangcandrew** [comment]
Otherwise, I think this bug fixes what it sets out to do.

**Pull request merged**